File: Kattis/hissingmicrophone/hissingmicrophone.cpp
#include <iostream>
#include <string>

int main() {
    std::string comp_string;
    std::cin >> comp_string;

    int comp_string_length = comp_string.length();

    for (int i = 0; i < comp_string_length; i++) {
        if (comp_string[i] == 's') {
            if (comp_string[i + 1] == 's') {
                std::cout << "hiss\n";
                return 0;
            }
            else {
                i++;
            }
        }
    }
    
    std::cout << "no hiss\n";
    return 0;
}

File: Kattis/hissingmicrophone/hissingmicrophone.py
comp_string = input()
comp_string_length = len(comp_string)
hissed = False

for i in range(comp_string_length):
    if comp_string[i] == 's':
        if i < comp_string_length - 1 and comp_string[i + 1] == 's':
            print("hiss")
            hissed = True
            break
        else:
            i += 1

if not hissed:
    print("no hiss")